src/ServiceControl.Audit/Auditing/AuditIngestion.cs

```diff
@@ -52,15 +52,27 @@ public AuditIngestion(
 
             errorHandlingPolicy = new AuditIngestionFaultPolicy(failedImportsStorage, settings.LoggingSettings, OnCriticalError);
 
-            watchdog = new Watchdog("audit message ingestion", EnsureStarted, EnsureStopped, ingestionState.ReportError, ingestionState.Clear, settings.TimeToRestartAuditIngestionAfterFailure, logger);
-
-            ingestionWorker = Task.Run(() => Loop(), CancellationToken.None);
+            watchdog = new Watchdog(
+                "audit message ingestion",
+                EnsureStarted,
+                EnsureStopped,
+                ingestionState.ReportError,
+                ingestionState.Clear,
+                settings.TimeToRestartAuditIngestionAfterFailure,
+                logger
+            );
         }
 
-        public Task StartAsync(CancellationToken _) => watchdog.Start(() => applicationLifetime.StopApplication());
+        public async Task StartAsync(CancellationToken cancellationToken)
```

Comment:
@andreasohlund @danielmarbach Alternative is to use BackgroundService.ExecuteAsync, assuming we can safely terminate ASAP:
```c#
protected override async Task ExecuteAsync(CancellationToken stoppingToken)
{
    await watchdog.Start(() => applicationLifetime.StopApplication());
    await Loop(stoppingToken);
    // Intentionally not invoking the following to shut down ASAP
    // watchdog.Stop();
    // channel.Writer.Complete();
    // if (transportInfrastructure != null)
    // {
    //     await transportInfrastructure.Shutdown(stoppingToken);
    // }
}
```

As everything is at-least-once processing and idempotent this would be the fastest method to shutdow.

Reply:
Yes, I already tested that and it works. As the loop pretty much immediately does IO its not affected which is why I removed the `Task.Run`. I can restore it for safety or add a code comment. Any preference @danielmarbach ?